lib/mix/tasks.ex

```diff
@@ -58,13 +58,28 @@ defmodule Mix.Tasks.Coveralls do
   end
 
   defp parse_common_options(args, options) do
-    {common_options, _remaining, unknown_args} = OptionParser.parse(args,
-      strict: [filter: :string, umbrella: :boolean, verbose: :boolean, pro: :boolean, parallel: :boolean],
-      aliases: [f: :filter, u: :umbrella, v: :verbose])
-    {
-      Keyword.keys(unknown_args),
-      Keyword.merge(options, common_options)
-    }
+    switches = [filter: :string, umbrella: :boolean, verbose: :boolean, pro: :boolean, parallel: :boolean]
+    aliases = [f: :filter, u: :umbrella, v: :verbose]
+    {common_options, _remaining, _invalid} = OptionParser.parse(args, switches: switches, aliases: aliases)
+
+    # the switches that excoveralls supports
+    supported_switches = Enum.map(Keyword.keys(switches), fn(s) -> "--#{s}" end)
```

Comment:
equivalent to `supported_switches = ~w(--filter --umbrella --verbose --pro --parallel -f -u -v)`